src/item_data.rs

```diff
@@ -0,0 +1,216 @@
+use std::{iter, ops::Range, path::PathBuf, rc::Rc};
+
+use ratatui::text::{Line, Span};
+
+use crate::{config::Config, git::diff::Diff, gitu_diff::Status, highlight};
+
+#[derive(Clone, Debug)]
+pub(crate) enum ItemData {
+    Raw(String),
+    AllUnstaged(usize),
+    AllStaged(usize),
+    AllUntracked(Vec<PathBuf>),
+    Reference {
+        prefix: &'static str,
+        kind: RefKind,
+    },
+    Commit {
+        oid: String,
+        short_id: String,
+        associated_references: Vec<RefKind>,
+        summary: String,
+    },
+    File(PathBuf),
+    Delta {
+        diff: Rc<Diff>,
+        file_i: usize,
+    },
+    Hunk {
+        diff: Rc<Diff>,
+        file_i: usize,
+        hunk_i: usize,
+    },
+    HunkLine {
+        diff: Rc<Diff>,
+        file_i: usize,
+        hunk_i: usize,
+        line_i: usize,
+        line_range: Range<usize>,
+    },
+    Stash {
+        message: String,
+        commit: String,
+        id: usize,
+    },
+    Header(SectionHeader),
+    BranchStatus(String, usize, usize),
+    Error(String),
+}
+
+impl Default for ItemData {
+    fn default() -> Self {
+        ItemData::Raw(String::new())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) enum RefKind {
+    Tag(String),
+    Branch(String),
+    Remote(String),
+}
+
+#[derive(Clone, Debug)]
+pub(crate) enum SectionHeader {
+    Remote(String),
+    Tags,
+    Branches,
+    NoBranch,
+    OnBranch(String),
+    UpstreamGone(String),
+    Rebase(String, String),
+    Merge(String),
+    Revert(String),
+    Stashes,
+    RecentCommits,
+    Commit(String),
+}
+
+impl ItemData {
```

Comment:
I thought if this before, but now that you do need the `id` of an item - it feels even more relevant.
What if:
```suggestion
impl Item {
```

Then you could skip the item_id argument! :)